src/python/tests/database/test_factory.py

```diff
@@ -196,3 +196,106 @@ def _format_db_data(server_url: URL, dbs: List[str], brc_mode: bool = False) ->
     result = factory.get_core_dbs_metadata(URL.create("mysql"), db_list=db_list)
     assert not DeepDiff(result, output)
+
+
+@pytest.mark.parametrize(
+    "arg_list, expected",
+    [
+        param(
+            ["--host", "localhost", "--port", "42", "--user", "me"],
+            {
+                "host": "localhost",
+                "port": 42,
+                "user": "me",
+                "password": None,
+                "url": make_url("mysql://me@localhost:42"),
+                "prefix": "",
+                "build": None,
+                "version": None,
+                "db_regex": "",
+                "db_list": None,
+                "brc_mode": False,
+                "log_level": "WARNING",
+            },
+            id="Default args",
+        ),
+        param(
+            [
+                "--host",
+                "localhost",
+                "--port",
+                "42",
+                "--user",
+                "me",
+                "--password",
+                "secret",
+                "--prefix",
+                "pre",
+                "--build",
+                "70",
+                "--version",
+                "114",
+                "--db_regex",
+                "%_core_%",
+                "--db_list",
+                __file__,
+                "--brc_mode",
+            ],
+            {
+                "host": "localhost",
+                "port": 42,
+                "user": "me",
+                "password": "secret",
+                "url": make_url("mysql://me:secret@localhost:42"),
+                "prefix": "pre",
+                "build": 70,
+                "version": 114,
+                "db_regex": "%_core_%",
+                "db_list": __file__,
+                "brc_mode": True,
+                "log_level": "WARNING",
+            },
+            id="New arg values",
+        ),
+    ],
+)
+def test_parse_args(arg_list: list[str], expected: dict) -> None:
+    """Tests the `factory.parse_args()` function."""
+    args = factory.parse_args(arg_list)
+    if args.db_list:
+        # DeepDiff is not able to compare two objects of Path type, so convert it to string
+        setattr(args, "db_list", str(args.db_list))
+    assert not DeepDiff(vars(args), expected)
+
+
+@pytest.mark.parametrize(
+    "arg_list, server_url, stdout",
+    [
+        (
+            ["--host", "localhost", "--port", "42", "--user", "me"],
+            make_url("mysql://me@localhost:42"),
+            '{\n    "test": "output"\n}\n',
+        ),
+    ],
+)
+@patch("ensembl.io.genomio.database.factory.get_core_dbs_metadata")
+def test_main(
+    mock_get_core_dbs_metadata: Mock,
+    capsys: CaptureFixture[str],
+    arg_list: list[str],
+    server_url: URL,
+    stdout: str,
+) -> None:
+    """Tests the `factory.main()` function (entry point).
+
+    Fixtures: capsys
+    """
+    mock_get_core_dbs_metadata.return_value = {"test": "output"}
```

Comment:
I think Im following what this test does, but would like a little help understanding at next dev meeting